test/channel_test.dart

```diff
@@ -8,4 +8,49 @@ void main() {
     channel.sendJoin(const Duration(seconds: 5));
     expect(channel.isJoining(), isTrue);
   });
+
+  group('on', () {
+    late RealtimeSubscription channel;
+
+    setUp(() {
+      channel = RealtimeSubscription('topic', RealtimeClient('endpoint'));
+    });
+
+    test('sets up callback for event', () {
+      var callbackCalled = 0;
+      channel.on('event', (dynamic payload, {String? ref}) => callbackCalled++);
+
```

Comment:
I feel like there is a better way to implement counting how many times a callback has been called, but I couldn't find a library to do spy in Dart. If someone has suggestions they are more than welcome!

Reply:
no worry, @dshukertjr. your code is simple and easy to understand. That's good enough.

Reply:
Thank you @phamhieu ! I will merge it right now!